File: framework/web/src/main/java/top/bettercode/simpleframework/web/serializer/KilogramSerializer.java
package top.bettercode.simpleframework.web.serializer;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.annotation.JacksonStdImpl;
import com.fasterxml.jackson.databind.ser.std.NumberSerializer;
import java.io.IOException;
import java.math.BigDecimal;
import top.bettercode.simpleframework.support.KilogramUtil;

@JacksonStdImpl
public class KilogramSerializer extends NumberSerializer {

  private static final long serialVersionUID = 1L;
  private static int newScale = 3;

  public KilogramSerializer() {
    super(Long.class);
  }

  public static void setNewScale(int newScale) {
    KilogramSerializer.newScale = newScale;
  }

  @Override
  public void serialize(Number value, JsonGenerator gen,
      SerializerProvider provider) throws IOException {
    gen.writeString(KilogramUtil.toKilogram(new BigDecimal(value.toString()), newScale).toString());
  }

}

File: web/src/main/kotlin/top/bettercode/summer/web/serializer/KilogramSerializer.kt
package top.bettercode.summer.web.serializer

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.databind.annotation.JacksonStdImpl
import com.fasterxml.jackson.databind.ser.std.NumberSerializer
import top.bettercode.summer.web.support.KilogramUtil.toKilogram
import java.math.BigDecimal

@JacksonStdImpl
class KilogramSerializer : NumberSerializer(Long::class.java) {
    override fun serialize(value: Number, gen: JsonGenerator,
                           provider: SerializerProvider) {
        gen.writeString(toKilogram(BigDecimal(value.toString()), newScale).toString())
    }

    companion object {
        private const val serialVersionUID = 1L
        private var newScale = 3
        fun setNewScale(newScale: Int) {
            Companion.newScale = newScale
        }
    }
}